src/common/components/PageLayout.jsx

```diff
@@ -88,7 +90,7 @@ const PageLayout = ({ menu, breadcrumbs, children }) => {
           <PageTitle breadcrumbs={breadcrumbs} />
         </Toolbar>
         <Divider />
-        {menu}
+        {menuMemo}
```

Comment:
Shouldn't the same thing be done in mobile layout?

Reply:
Also, how does it actually prevent re-rendering? Because the menu is still passed as a parameter every time. Have you verified that it actually works?

Reply:
It does not render again because in this case the useMemo has an empty dependency [ ].
I share the video of the tests.

## Video
https://github.com/traccar/traccar-web/assets/92123185/bc492c2b-c44e-495d-892a-a7d52fdd9fb3

Reply:
The menu won't be re-rendered, but the page will, no?

I'm not sure I understand what I'm supposed to see on the video, but I see re-rendering around 4th second.

Reply:
But I don't understand what I'm supposed to see in the video.

Reply:
Previously all items in the report navigation menu were rendered with each socket update.   With useMemo it no longer happens.

Reply:
I thought the problem with re-rendering the map, not the menu. Why would re-rendering the menu a problem?

Reply:
I thought that was what you were referring to in the comment https://github.com/traccar/traccar-web/pull/1251#issuecomment-2197750734

I agree with you, menu rendering is not a problem.


Also thank you for your time, because these days I am understanding your platform better.